src/lightning/pytorch/core/saving.py

```diff
@@ -87,7 +86,15 @@ def _load_from_checkpoint(
     if issubclass(cls, pl.LightningDataModule):
         return _load_state(cls, checkpoint, **kwargs)
     if issubclass(cls, pl.LightningModule):
-        return _load_state(cls, checkpoint, strict=strict, **kwargs)
+        if map_location is None:
+            map_location = list(checkpoint["state_dict"].values())[0].device
```

Comment:
I think this is fine because the framework imposes the existence of this key, but we should avoid the index error if there are no parameters at all

Reply:
Yep that looks better alright, thanks!
Do I need to include a check that there are parameters then?

Reply:
```suggestion
            assert len(checkpoint['state_dict']) > 0
            map_location = list(checkpoint["state_dict"].values())[0].device
```